packages/core/src/codewhisperer/service/transformByQ/transformApiHandler.ts

```diff
@@ -848,36 +847,26 @@ export function findDownloadArtifactStep(transformationSteps: TransformationStep
     }
 }
 
-interface IDownloadResultArchiveParams {
-    jobId: string
-    downloadArtifactId: string
-    pathToArchive: string
-}
-export async function downloadResultArchive({
-    jobId,
-    downloadArtifactId,
-    pathToArchive,
-}: IDownloadResultArchiveParams) {
+export async function downloadResultArchive(
+    jobId: string,
+    downloadArtifactId: string | undefined,
+    pathToArchive: string,
+    downloadArtifactType: TransformationDownloadArtifactType
+) {
     let downloadErrorMessage = undefined
     const cwStreamingClient = await createCodeWhispererChatStreamingClient()
+
     try {
         await downloadExportResultArchive(
             cwStreamingClient,
             {
                 exportId: jobId,
                 exportIntent: ExportIntent.TRANSFORMATION,
-                exportContext: {
```

Comment:
Just curious, why remove this?

Reply:
There's an issue in the proxy service - even though the values inside exportContext are marked nullable, if they're both null (which is the case when there's a prebuild error as `downloadArtifactId` is null), there's a `ValidationException` thrown. Right now, these values aren't needed, so I removed them. 

An alternative would be to optionally add `transformationExportContext` if `downloadArtifactId` isn't null -- I'm fixing up some things, so this will go into my next PR